slackclient/server.py

```diff
@@ -130,7 +130,17 @@ def rtm_connect(self, reconnect=False, timeout=None, use_rtm_start=True, **kwarg
         reply = self.api_requester.do(self.token, connect_method, timeout=timeout, post_data=kwargs)
 
         if reply.status_code != 200:
-            raise SlackConnectionError(reply=reply)
+            if 'depth' in kwargs:
+                depth = kwargs["depth"]
+            else:
+                depth = 0
+            if depth < 10 and reply.status_code == 429:
+                ratelimit_timeout = int(reply.headers.get('retry-after', 120))
+                logging.debug("HTTP 429: Rate limited. Retrying in %d seconds", ratelimit_timeout)
+                time.sleep(ratelimit_timeout)
+                self.rtm_connect(reconnect=reconnect, timeout=timeout, depth=depth + 1)
```

Comment:
oh i see, its used to measure the number of recursions. if this is strictly for private use, then i guess it doesn't need to be documented.